src/control/server/mgmt_system.go

```diff
@@ -209,7 +271,23 @@ func (svc *mgmtSvc) join(ctx context.Context, req *mgmtpb.JoinReq, peerAddr *net
 		Incarnation:             req.Incarnation,
 		CheckMode:               req.CheckMode,
 		ReplaceMode:             req.ReplaceMode,
-	})
+	}
+
+	if req.ReplaceMode {
+		rankToReplace, err := svc.membership.FindRankFromJoinRequest(joinReq)
+		if err != nil {
+			return nil, errors.Wrap(err, "join: replace existing rank")
+		}
+		if rankToReplace == ranklist.NilRank {
+			return nil, FaultJoinReplaceRankNotFound(joinReq)
+		}
```

Comment:
Maybe the "not found" fault should come from `svc.membership.FindRankFromJoinRequest(joinReq)`...

Reply:
done